### PR TITLE
Add guard against destroyed service

### PR DESCRIPTION
**Pull request by job13er**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG
 * **Added** a guard against `frost-navigation` service being destroyed when timeout finishes (attempting to fix [#204](https://github.com/ciena-frost/ember-frost-navigation/issues/204))

### CODE REVIEW

File: addon/components/nav-category.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ export default Component.extend({
       frostNavigation.dismiss()
       if (name !== active) { // click on another tab
         run.later(() => {
-          if (this.isDestroying || this.isDestoryed) {
+          if (this.isDestroying || this.isDestoryed || frostNavigation.isDestroying || frostNavigation.isDestroyed) {
```
Comment (srowhani) [inline addon/components/nav-category.js]: There is a typo in `this.isDestroyed`, would you mind fixing that.

Comment (job13er) [inline addon/components/nav-category.js]: thx

Comment (job13er) [inline addon/components/nav-category.js]: I guess when  I call `destroy()` it's only going into `isDestroying` and not `isDestroyed` or my test would have failed.

**Comment by job13er** [comment]
Approved

**Pull request merged**